packages/bcer-retailer-app/app/src/hooks/useCsvValidator.ts

```diff
@@ -1,30 +1,47 @@
 import React, { useState } from 'react';
 import * as yup from 'yup';
+import { useAxiosGet } from './axios';
+import Axios  from 'axios'
 
 export const useCsvValidator = () => {
   const [errors, setErrors] = useState<Array<{row: string, field: string, message: string}>>()
   const [validatedData, setValidatedData] = useState<Array<any>>()
+  const [{ data, error, loading }, getSuggestions] = useAxiosGet('', { manual: true })
 
   return {
     errors,
     validatedData,
-    validateCSV: (validationSchema: yup.ObjectSchema<any>, uploadData: Array<any>) => {
+    validateCSV: async(validationSchema: yup.ObjectSchema<any>, uploadData: Array<any>) => {
       let errorArray: Array<any> = [];
       let validatedDataArray: Array<any> = [];
+      
+      await Promise.all(uploadData.map(async(element, index) => {
 
-      uploadData.map((element, index) => {
         element.error = undefined;
         try {
-          const validatedDto = validationSchema.validateSync(element, { abortEarly: false });
-          validatedDataArray.push(validatedDto);
+          const validatedDto = await validationSchema.validateSync(element, { abortEarly: false });
+
+          try {
+            const {data} = await Axios.get(`https://geocoder.api.gov.bc.ca/addresses.json?minScore=70&maxResults=1&echo=false&autoComplete=false&brief=false&matchPrecision=occupant,unit,site,civic_number,block&addressString=${validatedDto.addressLine1}`)
+            // Features prop will only ever have length 0 or 1
+
+            if (data.features.length === 0 || data.features[0]?.properties.precisionPoints < 70) {
+              errorArray.push({row: index + 2, field: 'Geocoder Error', message: 'Google was unable to find a match. please edit the location and select the location address'})
+              element.error = true;
+            }
+          } catch (requestError) {
+            console.log(requestError)

```

Comment:
console

Reply:
Leaving this console log as it's purpose is to give some sort of a readout for the error and to not have an empty catch. The error is not necessarily user relevant but should still be shown somewhere in the event that there is ever an issue that needs debugging